CMakeLists.txt

```diff
@@ -38,7 +37,8 @@ else ()
 endif()
 
 set(CMAKE_C_FLAGS "${KVDK_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "${KVDK_COMPILE_FLAGS}")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${KVDK_COMPILE_SHARED_FLAGS}")
+set(CMAKE_CXX_FLAGS_RELEASE "${KVDK_COMPILE_FLAGS}")
```

Comment:
I think we have too many compile flags definition

Reply:
CMAKE_CXX_FLAGS, CMAKE_CXX_FLAGS_DEBUG, and CMAKE_CXX_FLAGS_RELEASE are default flags used by cmake.
When you set CMAKE_CXX_FLAGS, cmake will append parameters to it to make CMAKE_CXX_FLAGS_DEBUG and CMAKE_CXX_FLAGS_RELEASE. 
For example, you will see -O2 and -O3 appearing at the same time when build the release build because we set -O2 to CMAKE_CXX_FLAGS and cmake adds -O3 to CMAKE_CXX_FLAGS_RELEASE.

Reply:
https://stackoverflow.com/questions/60622643/why-my-cmake-does-not-add-cmake-cxx-flags-debug-to-cmake-cxx-flags